### ISSUE TITLE
Impact: Removal of `function.displayName` in Chromium 

### ISSUE THREAD

**Issue by chybie** [facebook/react#20838]
Chromium plan to [remove the non-standard property `fn.displayName` property](https://docs.google.com/document/d/1_GxIGRvv8ATp3GMhA3jUfVFxGKMWEkFWvtqsSS7QQ2M/edit#heading=h.7nki9mck5t64). How would this impact React [`component.displayName`](https://reactjs.org/docs/react-component.html#displayname)?

Links:
- Chromium: [Remove non-standard fn.displayName support](https://docs.google.com/document/d/1_GxIGRvv8ATp3GMhA3jUfVFxGKMWEkFWvtqsSS7QQ2M/edit#heading=h.7nki9mck5t64)
- React: [Component.displayName](https://reactjs.org/docs/react-component.html#displayname)

#### TL;DR;

Chromium has traditionally supported the non-standard `fn.displayName` property as a way to provide developer-controlled debug names for functions that show up in `error.stack` and in `DevTools` stack traces. 

With ES6, `fn.name` is configurable as a standard, which is a portable replacement for `fn.displayName`. Also, V8 added logic to automatically infer good function names from the source code.

**Comment by gaearon** [comment]
What happens in old browsers (e.g. IE) when you assign to `.name`? Does it throw?

**Comment by chybie** [comment]
Chrome supports `fn.name` since  Chrome 15 (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/displayName). IE doesn't even support `fn.displayName`... 

So the assumption is older browser debugging would just be as it is.

**Comment by gaearon** [comment]
To clarify, I'm asking if there are any environments where `name` exists but is non-configurable.

**Comment by mrsimb** [comment]
> To clarify, I'm asking if there are any environments where `name` exists but is non-configurable.

in chrome, you can't change fn.name
so if there's any hoc or other decorator function that has its own name, without fn.displayName you won't see a proper component name in react-developer-tools tree

**Comment by chybie** [comment]
In the case which it's non-configurable, then browsers will throw the best name it can find. In some case, it could be `anonymous` - which is not desired.
